src/annotator/config/settings.js

```diff
@@ -20,9 +21,21 @@ import isBrowserExtension from './is-browser-extension';
  * @return {SettingsGetters}
  */
 export default function settingsFrom(window_) {
-  const jsonConfigs = parseJsonConfig(window_.document);
+  // Prioritize the `window.hypothesisConfig` function over the JSON format
+  // Via uses `window.hypothesisConfig` and makes it non-configurable and non-writable.
+  // In addition, Via sets the `ignoreOtherConfiguration` option to prevent configuration merging.
   const configFuncSettings = configFuncSettingsFrom(window_);
 
+  let jsonConfigs;
+  if (
+    configFuncSettings.hasOwnProperty('ignoreOtherConfiguration') &&
+    toBoolean(configFuncSettings.ignoreOtherConfiguration) === true
```

Comment:
Assuming that `toBoolean` always returns a boolean as the name implies the `=== true` part is not strictly needed. Also it it casts undefined => boolean then the `hasOwnProperty` part is not strictly needed either.

Reply:
That's true. The only reason to use `hasOwnProperty` would be to avoid prototype pollution.